src/mono/mono/metadata/bundled-resources.c

```diff
@@ -126,8 +126,9 @@ bundled_resources_resource_id_hash (const char *id)
 void
 mono_bundled_resources_add (MonoBundledResource **resources_to_bundle, uint32_t len)
 {
-	MonoDomain *domain = mono_get_root_domain ();
-	g_assert (!domain);
+	// TODO FIXME, why this was necessary ?
```

Comment:
It looks like this check was only a year or so old, maybe the person who added it still remembers. My guess is it was something like "make sure we don't accidentally load satellite assemblies too late, since the app would run but misbehave without them"

Reply:
yea that sounds right. it's probably a cheap way to ensure this runs before the runtime gets too far in initialization.